tools/benchmark/src/Configuration.ts

```diff
@@ -92,34 +92,58 @@ for (const type of Object.values(TestType)) {
  * Arguments to `benchmark`
  * @public
  */
-export type BenchmarkArguments = BenchmarkSyncArguments | BenchmarkAsyncArguments;
+export type BenchmarkArguments = Titled & (BenchmarkSyncArguments | BenchmarkAsyncArguments);
+
+export type BenchmarkRunningOptions = (BenchmarkSyncArguments | BenchmarkAsyncArguments) &
+	BenchmarkTimingOptions &
+	OnBatch &
+	HookArguments;
+
+export type BenchmarkRunningOptionsSync = BenchmarkSyncArguments & BenchmarkTimingOptions & OnBatch;
+
+export type BenchmarkRunningOptionsAsync = BenchmarkAsyncArguments &
+	BenchmarkTimingOptions &
+	OnBatch;
 
 /**
- * Arguments to benchmark a synchronous function
+ * Object with a "title".
  * @public
  */
-export interface BenchmarkSyncArguments extends BenchmarkOptions {
+export interface Titled {
 	/**
 	 * The title of the benchmark. This will show up in the output file, well as the mocha reporter.
 	 */
 	title: string;
+}
 
+/**
+ * Arguments to benchmark a synchronous function
+ * @public
+ */
+export interface BenchmarkSyncArguments extends BenchmarkSyncFunction, BenchmarkOptions {}
```

Comment:
Unnecessary to extend `BenchmarkOptions` here since `BenchmarkSyncFunction` already does? Same for `BenchmarkAsyncArguments`.